Feature GCM: Push Notifications

lennart-bader opened the pull request:
Hier der PR dazu - ist aber noch in Arbeit.

Siehe #25 dazu.

lennart-bader committed:
Added settings strings for gcm

lennart-bader committed:
Added settings strings for gcm

lennart-bader committed:
Added loading time caching.
Applied to Notification Fragment.

lennart-bader committed:
Added loading time caching.
Applied to Notification Fragment.

lennart-bader committed:
Added GCM Services

lennart-bader committed:
Added GCM Services

lennart-bader committed:
Added sendGcmToken to MyTFG class.
Replaced deprecated getDrawable Method.

lennart-bader committed:
Added sendGcmToken to MyTFG class.
Replaced deprecated getDrawable Method.

lennart-bader committed:
Added GCM support to AndroidManifest and Grade build files.

lennart-bader committed:
Added GCM support to AndroidManifest and Grade build files.

lennart-bader committed:
Added GCM initialisation

lennart-bader committed:
Added GCM initialisation

lennart-bader committed:
Added GCM token resend on Login

lennart-bader committed:
Added GCM token resend on Login

lennart-bader committed:
Added google-services.json

lennart-bader committed:
Added google-services.json

Turakar commented on app/src/main/java/de/mytfg/app/android/MyTFG.java:
```diff
@@ -123,4 +128,24 @@ public static void logout() {
+                MytfgApi.call("ajax_gcm_register", params, callback);
+            }
+        }
+    }
```
Werden Fälle, wo die Methode aufgerufen wird ohne eingeloggt zu sein, hier nicht einfach ignoriert?

rleh committed:
GCM: fix travis build / update gms lib to 8.1.0 and add google play services to dependencies

rleh committed:
GCM: fix travis build / update gms lib to 8.1.0 and add google play services to dependencies

rleh committed:
revert gms lib to 7.8.0

rleh committed:
revert gms lib to 7.8.0

rleh commented:
Mit Google Play Services Version 7.8.0 scheint die App korrekt zu Builden.
Jedoch scheitert der Build an einem Out-Of-Memory Error, der ist jedoch im aktuellen develop branch bereits gefixt.

Turakar committed:
Merge pull request #29 from MyTFG/feature_notifications

Add date/time text to notifications view

lennart-bader committed:
Fixed GCM notifications not showing.
Added GCM Manager to group Notifications.
Added GCM Manager to handle callbacks for modules.

lennart-bader committed:
Fixed merge conflicts

lennart-bader committed:
Added callback Method for GCM: onReceive and onClick available.

lennart-bader commented:
Änderungen:
- Benachrichtigungen werden wieder angezeigt
- Benachrichtigungen werden (wie in MyTFG) gruppiert
- Module können onClick und onReceive callbacks definieren (siehe GcmCallbackRegistration)
- Die Callbacks liefern ein GcmNotification Objekt, welches die notwendigen Informationen enthält:
  - Type: Der "Typ" bzw das Modul
  - Grouper: Der Grouper für die Notification. daraus lässt sich beispielsweise die Konversation entnehmen
  - Message
  - Title

lennart-bader commented:
Spricht hier was gegen Mergen?

lennart-bader committed:
Added possibility to hide Notifications.

rleh commented:
Von mir aus nicht :+1:

rleh commented:
Oh sorry, habe aus versehen auf _Comment and close_ gedrückt.
